src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -95,6 +96,21 @@ public void Dispose()
                     s_blendedNodesPool.ForgetTrackedObject(blendedTokens);
                 }
             }
+
+            var lexedTokens = _lexedTokens;
+            if (lexedTokens != null)
+            {
+                _lexedTokens = null;
+                if (lexedTokens.Length < 4096)
```

Comment:
I don't have a clear sense of how to choose this magic number. I guess this is fine and we can tweak later if needed.

Reply:
Some back-of-the-napkin math suggests that, at a guesstimated 6-10 tokens/line, this will cover C# files around 4-600 lines in length. That does feel somewhat low to me for real-world scenarios (since I think I might be low-balling the number of tokens/line on average), but I agree we can tweak if needed. Might be worth a comment indicating such though.